api/src/main/java/com/messagebird/MessageBirdClient.java

```diff
@@ -1848,4 +1854,168 @@ public String downloadFile(String id, String filename, String basePath) throws G
         final String url = String.format("%s%s/%s", MESSAGING_BASE_URL, FILES_PATH, id);
         return messageBirdService.getBinaryData(url, basePath, filename);
     }
-} 
+
+    /****************************************************************************************************/
+    /** WhatsApp Templates                                                                             **/
+    /****************************************************************************************************/
+
+    /**
+     * Create a WhatsApp message template through messagebird.
+     *
+     * @param template {@link WhatsAppTemplate} object to be created
+     * @return {@link WhatsAppTemplateResponse} response object
+     * @throws UnauthorizedException if client is unauthorized
+     * @throws GeneralException      general exception or invalid template format
+     */
+    public WhatsAppTemplateResponse createWhatsAppTemplate(final WhatsAppTemplate template)
+        throws UnauthorizedException, GeneralException {
+        template.validate();
```

Comment:
This `validate` method only checks `components` field and return. `name`, `language`, `category`, `components` are required fields to create a template. Can you check other fields too? Also, Can you return an IllegalException if they are null with a meaningful message like you did on other endpoints?